### PR TITLE
Fix set_vring_addr issues

### PR DESCRIPTION
**Pull request by justxuewei**
`VhostBackend::set_vring_addr()` receives a vring config data which
contains the addresses of desc table, used ring and avail ring.
`VhostBackend::is_valid()` checks the addresses in the guest address space.
`VHOST_SET_VRING_ADDR` uses the addresses in the host address space.
However, the method doesn't convert those addresses.

To address this issue, the addresses passed by the config are checked in
the guest address space. Then, they are converted by
`VringConfigData::to_vhost_vring_addr()` into the host address space to
setup the vring on the kernel.

Signed-off-by: Xuewei Niu <niuxuewei.nxw@antgroup.com>

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,7 @@
 /build
 /kcov_build
 /target
+.vscode
```
Comment (ZizhengBian) [inline .gitignore:4]: It's better to split this line into another patch.

Comment (justxuewei) [reply .gitignore]: Fixed

**Review by stefano-garzarella — DISMISSED** [review]
LGTM!

Maybe we should add a test to cover this scenario.

**Review by eryugey — APPROVED** [review]

File: crates/vhost/src/backend.rs [hunk 2]
```diff
@@ -57,6 +61,35 @@ impl VringConfigData {
             0
         }
     }
+
+    #[cfg(feature = "vhost-kern")]
```
Comment (jiangliu) [inline crates/vhost/src/backend.rs:65]: Can we move this changes into `src/vhost_kern/mod.rs`, so we can avoid those condition compilations.

Comment (justxuewei) [reply crates/vhost/src/backend.rs]: I moved the method into `src/vhost_kern/mod.rs`, Please take a look, thanks.

**Review by stefano-garzarella — APPROVED** [review]

**Review by jiangliu — APPROVED** [review]
